### PR TITLE
feat: implement robust @if instruction and tests

### PR DESCRIPTION
**Pull request by luffy-orf**
### Summary

This PR implements the `QIf` class for OpenQASM v2.0, enabling robust quantum conditional execution as described in [issue #43](https://github.com/hhat-lang/hhat_lang/issues/43).

### Details

- Implements the `QIf` class, inheriting from `QInstr`, following the model of `QRedim`
- Supports all built-in quantum types: `@bool`, `@u2`, `@u3`, `@u4`
- Handles multi-qubit conditions, multiple options (if-elif-else), and default/else branches
- Generates correct OpenQASM v2.0 code for all cases
- Adds comprehensive tests for:
  - Single and multi-bit conditions
  - Multiple options and else branches
  - Multiple instructions in the body
  - All-zero and all-one edge cases

### CODE REVIEW

File: python/tests/lowlevel/qlang/openqasm/v2/test_lowlevelqlang.py [hunk 1]
```diff
@@ -68,3 +69,98 @@ def test_gen_program_single_q0_redim() -> None:
...
+            2: lambda idxs, executor, **kwargs: qredim(idxs=(idxs[0],), **kwargs)[0],
+            3: lambda idxs, executor, **kwargs: qredim(idxs=(idxs[0],), **kwargs)[0],
+        },
+        cond_size=2
+    )
+    assert "measure q[0] -> c[0];" in code[0]
+    assert "measure q[1] -> c[1];" in code[1]
+    assert any("if (c[0]==1 && c[1]==0)" in line for line in code)  # 2 = 10b
+    assert any("if (c[0]==1 && c[1]==1)" in line for line in code)  # 3 = 11b
+
```
Comment (Doomsk) [inline python/tests/lowlevel/qlang/openqasm/v2/test_lowlevelqlang.py:109]: I don't think this is a valid format of `if`-statement for OpenQASMv2.0. Please check its arXiv paper: https://arxiv.org/pdf/1707.03429 

On page 21 it shows OpenQASM v2 grammar
On page 9 it shows a Open QASM language statement, including the if statement
On pages 14, 15, 16, 19 it shows code examples

File: python/src/hhat_lang/low_level/quantum_lang/openqasm/v2/instructions.py [hunk 2]
```diff
@@ -122,11 +122,85 @@ def __call__(
...
-
+        """Transforms `@if` instruction to openQASM v2.0 code (call with options)."""
         self._instr_status = InstrStatus.RUNNING
-        raise NotImplementedError()
+        code = []
+        cond_idxs = idxs[:cond_size]  # The indices of the qubits for the condition
+        body_idxs = idxs[cond_size:]  # The rest are for the body
+        # 1. Measure all condition qubits into corresponding classical bits
+        for qidx in cond_idxs:
+            code.append(f"measure q[{qidx}] -> c[{qidx}];")
```
Comment (Doomsk) [inline python/src/hhat_lang/low_level/quantum_lang/openqasm/v2/instructions.py:136]: The `@if` instruction is, as defined in #43 is a quantum instruction, handling quantum data, so it does not handle classical data at all. It is not a `if` statement (classical one).

Check out the issue description.

**Review by Doomsk — CHANGES_REQUESTED** [review]
@luffy-orf  there are some comments to address. Please read the issue carefully and try to address them properly.

**Comment by Hazarre** [comment]
Hi @Doomsk, I'd love to take a stab at this. 

Can you help me understand how to build circuits with IRs by writing out the equivalent to the following?
```
OPENQASM 2.0;
include "qelib1.inc";
qreg q[2];
creg c[2];

if(c[0]==1) x q[1];
measure q -> c;
```
